### ISSUE TITLE
Theme settings gets whiped after a certain amount of time

### ISSUE THREAD

**Issue by modmoto** [w3champions/website#373]
Every once in a while the theme setting in the local storage vanishes and you get undead as a default theme. Maybe related to a deployment?

**Comment by joleb** [comment]
For me, it never changed back to undead, its may related to your browser erasing local storage from time to time?

**Comment by modmoto** [comment]
I just have chrome and realized that this happened some times allready. Not a big deal just wanted to mark it here so when I have some time again, I might fix it ^^

**Comment by Cepheid-UK** [comment]
This happens for me on mobile (S20 android, chrome browser) but not on desktop strangely.

**Comment by modmoto** [comment]
It just got whiped for me again and i realized I was on www.w3champions.com. When I went without www. it was my old setting again. Maybe we should redirect www. to the normal site anyways? That might also be weird with the token for example. But my guess is, that it has something to do with that.

**Comment by joleb** [comment]
If that's the case, simply check your local storage 😄 
![image](https://user-images.githubusercontent.com/38313415/114622765-afa4fc80-9cae-11eb-8cd6-619bc9704508.png)
But redirecting from non-www to www is the best solution imo.

**Comment by modmoto** [comment]
yes i know, the local storage was empty on www but my guess is that I have been there mutliple times aswell and set it to my theme.

I just wanted this as an reminder and maybe someone has an idea why that happens.

**Comment by kontrarian** [comment]
Hey everyone,

Just checking in on this.

So we've confirmed that localStorage is different for each origin so you might get different results depending on whether you got to the tld or the www subdomain. @modmoto suggested that we do a redirect from www to the tld (or perhaps vice versa) which would solve the issue of having multiple localStorage's for the same user perceived site.

The localStorage [spec](https://www.w3.org/TR/webstorage/#dom-localstorage) indicates that the storage doesn't expire but could be cleared by the user or for deemed security reasons (whatever that means). Has anyone been able to confirm that localStorage has actually expired for the same domain on mobile or desktop?

**Comment by Cepheid-UK** [comment]
Not sure on that question about localStorage but I agree re-routing the URL is probably the best solution to this.

**Comment by MichaelDeitz** [comment]
Are we still planning to fix this by redirecting www.  to the main domain so localStorage stays consistent?

**Comment by gustav87** [comment]
Fixed by redirecting www to non-www.
